File: backend/routers/meals.py
from fastapi import APIRouter
from models.meals      import MealRequest
from services.meals_ai import generate_meals

router = APIRouter(prefix="/meals", tags=["meals"])

@router.post("")
async def get_meals(req: MealRequest):
    meals = await generate_meals(
        prompt     = req.prompt,
        max_time   = req.max_time   or 30,
        max_budget = req.max_budget or 150,
        num_meals  = req.num_meals  or 3,
    )
    return {"meals": meals}

File: frontend/src/api/meals.ts
import axios from "axios"

const BASE = "http://localhost:8000"

export interface Meal {
  name:                 string
  description:          string
  time_minutes:         number
  estimated_cost_inr:   number
  ingredients:          string[]
  steps:                string[]
  tags:                 string[]
}

export interface MealRequest {
  prompt?:     string
  max_time?:   number
  max_budget?: number
  num_meals?:  number
}

export const mealsApi = {
  generate: (req: MealRequest) => axios.post<{ meals: Meal[] }>(`${BASE}/meals`, req),
}